### PR TITLE
Adding loader for jamendo moodtheme

### PR DESCRIPTION
**Pull request by PRamoneda**
### Title

 Please use the following title: "Adding loader for MyDATASET". If your pull request is work in progress, change your title to "[WIP] Adding loader for MyDATASET" to avoid reviews while the loader is not ready.

### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the size of the datasets (e.g. number files and duration, hours)
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of music included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `mirdata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in mirdata, e.g. `mirdata/my_dataset.py`
- [x] Create tests for your loader in `tests/datasets/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/mirdata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

### CODE REVIEW

**Comment by PRamoneda** [comment]
hey!! ready to review!

**Review by philtgun — COMMENTED** [review]
- The `.download()` description tells to put the audio under `mtg_jamendo_autotagging_moodtheme/audio`, but the `.validate()` is checking under `audios` directory - an inconsistency. 
- The output of `.validate()` is huge when there are no tracks, not sure if such behavior is static across mirdata, but for such a big dataset is a bit too much output. It also doesn't say that the tracks are missing.
- Not sure when is the metadata downloaded - I put the audio files manually, successfully ran `.validate()`, but still can't access the tags

**Comment by magdalenafuentes** [comment]
Wow thanks @philtgun for the detailed review!

**Comment by PRamoneda** [comment]
Thank you so much for your review!!!  @philtgun . I am going to add your suggestions and errors now :)

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
+# -*- coding: utf-8 -*-
+"""MTG_jamendo_autotagging_moodtheme Dataset Loader
+
+.. admonition:: Dataset Info
+    :class: dropdown
+
+    The MTG Jamendo autotagging mood/theme Dataset is a new open dataset for music auto-tagging. It
+    is built using music available at Jamendo under Creative Commons licenses and tags provided by content uploaders. The
+    dataset contains over 19,000 full audio tracks with 195 tags from mood/theme. It is provided
```
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:9]: or better the full number 18,486

Done!

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+    relaxingmusic
+    trailermusic
+    lovemusic
+    joy
+    xmas
+    epicmusic
+    motivate
+    darkness
+    sciencefiction
+    festif
```
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:44]: done!

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+    def __init__(self, data_home=None, version="default"):
+        super().__init__(
+            data_home,
+            version,
+            name="mtg_jamendo_autotagging_moodtheme",
+            track_class=Track,
+            bibtex=BIBTEX,
+            download_info=DOWNLOAD_INFO,
+            indexes=INDEXES,
+            license_info="Creative Commons Attribution Share Alike 4.0 International.",
```
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:219]: NonCommercial Added!

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+        return {"metadata": meta, "splits": splits}
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    def split_load_tracks(self, num):
+        """Load a MTG_jamendo_autotagging_moodtheme pre-defined split. There are five different train/validation/tests splits.
+        Args:
+             num (int): split to be reatrieved from 0 to 4
```
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:341]: Oups! I am sorry! Done!

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+    address = "Long Beach, CA, United States",
+    url = "http://hdl.handle.net/10230/42015"
+}"""
+INDEXES = {
+    "default": "1.0",
+    "test": "1.0",
+    "1.0": core.Index(filename="mtg_jamendo_autotagging_moodtheme_index_1.0.json"),
+}
+DOWNLOAD_INFO = """
+    The audio files can be downloaded following the path described in https://multimediaeval.github.io/2019-Emotion-and-Theme-Recognition-in-Music-Task/
```
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:87]: ok!

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:9]: done!

File: docs/source/table.rst [hunk 6]
```diff
@@ -192,6 +192,14 @@
      - .. image:: https://licensebuttons.net/l/by-nc-sa/3.0/80x15.png
           :target: https://creativecommons.org/licenses/by-sa/3.0
 
+   * - MTG_jamendo_autotagging_moodtheme
+     - - audio: ✅
+       - annotations: ✅
+     - - moodtheme annotations
+     - >18000
+     - .. image:: https://licensebuttons.net/l/by-nc-sa/4.0/80x15.png
+          :target: https://creativecommons.org/licenses/by-sa/4.0
```
[the remark this answers is not in the document]
Comment (PRamoneda) [reply docs/source/table.rst]: done!

**Comment by PRamoneda** [comment]
Hi Philip! @philtgun 
>> The output of .validate() is huge when there are no tracks, not sure if such behavior is static across mirdata, but for such a big dataset is a bit too much output. It also doesn't say that the tracks are missing.

This is the usual behaviour... but you are right maybe a tiny general description would be better

**Comment by PRamoneda** [comment]
hi @philtgun good catch!
>> Not sure when is the metadata downloaded - I put the audio files manually, successfully ran .validate(), but still can't access the tags. Not sure if the metadata should be downloaded with .download(partial_download=['metadata']) which still just shows the instructions for downloading the audio and does nothing.

**Comment by PRamoneda** [comment]
It is caused because i havent run the full_dataset test. I am running it now!

**Comment by PRamoneda** [comment]
four hours later....

```Console
 % pytest -s tests/test_full_dataset.py --local --dataset mtg_jamendo_autotagging_moodtheme
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pep8.py:110: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
============================================================================== test session starts ===============================================================================
platform darwin -- Python 3.7.9, pytest-6.2.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/pedroramonedafranco/PycharmProjects/mirdata3
plugins: localserver-0.5.0, mock-3.3.1, cov-2.10.1, pep8-1.0.6
collected 4 items                                                                                                                                                                

tests/test_full_dataset.py If this dataset does not have openly downloadable data, follow the instructions printed by the download message and rerun this test.
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 18486/18486 [11:33<00:00, 26.67it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 18486/18486 [4:27:18<00:00,  1.15it/s]
..

================================================================================ warnings summary ================================================================================
tests/test_full_dataset.py: 18486 warnings
  /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/librosa/core/audio.py:162: UserWarning: PySoundFile failed. Trying audioread instead.
    warnings.warn("PySoundFile failed. Trying audioread instead.")

-- Docs: https://docs.pytest.org/en/stable/warnings.html
================================================================ 4 passed, 18486 warnings in 16739.36s (4:38:59) =================================================================
(base) 
```

**Comment by PRamoneda** [comment]
Ready to merge! @rabitt @magdalenafuentes

**Review by nkundiushuti — APPROVED** [review]

**Comment by PRamoneda** [comment]
hi! I would like to remove the dataset from my laptop. What is your opinion about this loader?

**Comment by rabitt** [comment]
Hey @PRamoneda I'll try to take a look tomorrow!

File: mirdata/datasets/good_sounds.py [hunk 7]
```diff
@@ -266,7 +266,7 @@ def to_jams(self):
 
 @io.coerce_to_bytes_io
 def load_audio(fhandle: BinaryIO) -> Tuple[np.ndarray, float]:
-    """Load a Beatles audio file.
+    """Load a GOOD-SOUNDS audio file.
```
Comment (rabitt) [inline mirdata/datasets/good_sounds.py:269]: good catch!

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+                > 00/
+                ...
+                > 99/
+"""
+REMOTES = {
+    "metadata": download_utils.RemoteFileMetadata(
+        filename="metadata.zip",
+        url="https://zenodo.org/record/3826813/files/data.zip?download=1",
+        checksum="039ce10f267f6e4e9f72837c76d72b2f",
+        destination_dir=".",
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:82]: nit: you don't need to specify destination_dir if it's `'.'` 
```suggestion
```

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+    corporate, dark, deep, documentary, drama, dramatic, dream, emotional, energetic, epic, fast, film, fun, funny,
+    game, groovy, happy, heavy, holiday, hopeful, horror, inspiring, love, meditative, melancholic, mellow, melodic,
+    motivational, movie, nature, party, positive, powerful, relaxing, retro, romantic, sad, sexy, slow, soft,
+    soundscape, space, sport, summer, trailer, travel, upbeat, uplifting.
+
+    Emotion and theme recognition is a popular task in music information retrieval that is relevant for music search and
+    recommendation systems.
+
+    This task involves the prediction of moods and themes conveyed by a music track, given the raw audio. The examples
+    of moods and themes are: happy, dark, epic, melodic, love, film, space etc. Each track is tagged with at least one
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:24]: is there a complete list available?

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+                            "TRACK_ID",
+                            "ARTIST_ID",
+                            "ALBUM_ID",
+                            "PATH",
+                            "DURATION",
+                            "TAGS",
+                        ],
+                    )
+                )
+                split["train"] = [m["TRACK_ID"] for m in d[1:]]
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:254]:
```suggestion
                reader = csv.reader(fhandle, delimiter="\t")
                track_uris = [line[0] for line in reader]
                split["train"] = track_uris[1:]
```

and the same for the lines below

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+                    "ALBUM_ID": m["ALBUM_ID"],
+                    "PATH": m["PATH"],
+                    "DURATION": m["DURATION"],
+                    "TAGS": m["TAGS"],
+                }
+                for m in d[1:]
+            }
+
+        splits = []
+        for ii in range(5):
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:230]:
```suggestion
        for split_number in range(5):
```

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 12]
```diff
@@ -0,0 +1,340 @@
...
+            split = {}
+            path_train = os.path.join(
+                self.data_home,
+                "data",
+                "splits",
+                "split-" + str(ii),
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:236]:
```suggestion
                f"split-{split_number}",
```

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 13]
```diff
@@ -0,0 +1,340 @@
...
+                            "ARTIST_ID",
+                            "ALBUM_ID",
+                            "PATH",
+                            "DURATION",
+                            "TAGS",
+                        ],
+                    )
+                )
+                split["test"] = [m["TRACK_ID"] for m in d[1:]]
+            splits.append(split)
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:301]: Maybe it's better if `splits` is a dictionary keyed by split_number:
```suggestion
            splits[split_number] = split
```

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 14]
```diff
@@ -0,0 +1,340 @@
...
+        return {"metadata": meta, "splits": splits}
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:302]:
```suggestion
        meta["splits"] = splits
        return meta
```

This way the suggested change about `_track_metadata` above will work!

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 15]
```diff
@@ -0,0 +1,340 @@
...
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    @core.cached_property
+    def artist_id(self) -> str:
+        return self._metadata()["metadata"][self.track_id]["ARTIST_ID"]
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:136]:
```suggestion
        return self._track_metadata.get("ARTIST_ID")
```

and the same for `album_id`, `duration` and `tags`

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 16]
```diff
@@ -0,0 +1,340 @@
...
+            meta = {
+                m["TRACK_ID"]: {
+                    "ARTIST_ID": m["ARTIST_ID"],
+                    "ALBUM_ID": m["ALBUM_ID"],
+                    "PATH": m["PATH"],
+                    "DURATION": m["DURATION"],
+                    "TAGS": m["TAGS"],
+                }
+                for m in d[1:]
+            }
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:227]:
```suggestion
        with open(meta_path, "r") as fhandle:
            reader = csv.reader(fhandle, delimiter="\t")
            # columns are track_id, artist_id, album_id, path, duration, tags
            track_metadata = {
                line[0]: {
                    "ARTIST_ID": line[1],
                    "ALBUM_ID": line[2],
                    "PATH": line[3],
                    "DURATION": line[4],
                    "TAGS": line[5]
                } 
                for line in reader
                if line[0] != "TRACK_ID"  # skip first line of csv file
            }
```

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 17]
```diff
@@ -0,0 +1,340 @@
...
+                )
+                split["test"] = [m["TRACK_ID"] for m in d[1:]]
+            splits.append(split)
+        return {"metadata": meta, "splits": splits}
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    def split_load_tracks(self, num):
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:308]:
```suggestion
    def get_track_ids_for_split(self, split_number):
```

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 18]
```diff
@@ -0,0 +1,340 @@
...
+            if k in meta["splits"][num]["validation"]
+        }
+
+        test_tracks = {
+            k: v
+            for k, v in self._index["tracks"].items()
+            if k in meta["splits"][num]["test"]
+        }
+
+        return train_tracks, validation_tracks, test_tracks
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:340]: if the dictionary change gets made above, this function gets much simpler! Also - I think it's better just to give the list of track uris, and not the tracks themselves


```suggestion
        Returns:
            * dict: {"train": [...], "validation": [...], "test": [...]} - the train split

        """
        if not (0 <= num <= 4):
            raise Exception("Splits avaiables from num 0 to 4")

    return self._metadata["splits"][split_number]
```

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 19]
```diff
@@ -0,0 +1,340 @@
+# -*- coding: utf-8 -*-
+"""MTG_jamendo_autotagging_moodtheme Dataset Loader
+
+.. admonition:: Dataset Info
+    :class: dropdown
+
+    The MTG Jamendo autotagging mood/theme Dataset is a new open dataset for music auto-tagging. It
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:7]: Can we include the website somewhere in this description? https://github.com/MTG/mtg-jamendo-dataset

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py [hunk 20]
```diff
@@ -0,0 +1,340 @@
+# -*- coding: utf-8 -*-
+"""MTG_jamendo_autotagging_moodtheme Dataset Loader
```
Comment (rabitt) [inline mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:2]: same comment here - does the official name have the underscores?

File: docs/source/table.rst [hunk 21]
```diff
@@ -191,7 +191,15 @@
        - :ref:`genre`
      - 9579
      - .. image:: https://licensebuttons.net/l/by-nc-sa/3.0/80x15.png
-          :target: https://creativecommons.org/licenses/by-sa/3.0
+          :target: https://creativecommons.org/licenses/by-nc-sa/3.0
+
+   * - MTG_jamendo_autotagging_moodtheme
```
Comment (rabitt) [inline docs/source/table.rst:196]: do we need the `_` here (is it the official name)?

**Review by rabitt — CHANGES_REQUESTED** [review]
New loader checklist:
- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`
- [x] Run the script on the canonical version of the dataset and save the index in `mirdata/indexes/` e.g. `my_dataset_index.json`
- [x] Create a module in `mirdata`, e.g. `mirdata/my_dataset.py`
- [x] Create tests in `tests/`, e.g. `test_my_dataset.py`
- [x] Add the module to `docs/source/mirdata.rst`
- [x] Add the module to `docs/source/table.rst`
- [x] Run test_full_dataset.py
- [ ] Make sure the docs build properly

File: mirdata/datasets/good_sounds.py (hunk 7 above)
Comment (PRamoneda) [reply mirdata/datasets/good_sounds.py:269]: done

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py (hunk 9 above)
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:24]: yes I add a link to there

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py (hunk 19 above)
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:7]: done

File: mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py (hunk 9 above)
Comment (PRamoneda) [reply mirdata/datasets/mtg_jamendo_autotagging_moodtheme.py:24]: done

File: docs/source/table.rst (hunk 21 above)
Comment (PRamoneda) [reply docs/source/table.rst:196]: done

**Review by rabitt — APPROVED** [review]

**Comment by rabitt** [comment]
Looks great @PRamoneda ! Merging now

**Pull request merged**